app/models/competition.rb

```diff
@@ -246,6 +250,23 @@ class Competition < ApplicationRecord
   validates :external_website, :external_registration_page, length: { maximum: MAX_URL_LENGTH }
   validates :contact, length: { maximum: MAX_MARKDOWN_LENGTH }
 
+  validate :validate_newcomer_reserved_spots, if: -> { competitor_limit.present? }
+  private def validate_newcomer_reserved_spots
+    max_newcomer_spots = (competitor_limit * MAX_NEWCOMER_SPOTS_RESERVED_FRACTION).floor
+    if newcomer_reserved_spots > max_newcomer_spots
+      errors.add(:newcomer_reserved_spots, 'cant reserve more than 50% of spots for newcomers')
+    end
+  end
+
+  def newcomers_competing_count
+    registrations.accepted.joins(:user).merge(User.newcomers).count
+  end
+
+  def newcomer_month_reserved_spots_remaining
+    newcomer_month_eligible_competitors = registrations.accepted.joins(:user).merge(User.newcomer_month_eligible).count
+    newcomer_reserved_spots - newcomer_month_eligible_competitors
+  end
```

Comment:
Can you move these into functions under `registration.rb` similar to the already existing `def accepted_count` and `def accepted_and_paid_pending_count`?

Reply:
In the latter case of `newcomer_month_reserved_spots_remaining`, it will suffice to only move the count itself into `registration.rb` and leave the subtraction here

Reply:
Moved!